src/assembly/local/quadrature_table.rs

```diff
@@ -101,6 +101,10 @@ where
 
         Self { points, weights, data }
     }
+
+    pub fn into_parts(self) -> (NestedVec<OPoint<T, GeometryDim>>, NestedVec<T>, NestedVec<Data>) {
+        (self.points, self.weights, self.data)
+    }
```

Comment:
unfortunately the library is not quite consistent here: we use the `(weights, points)` ordering in many other places. In a generic context, the order here is not a big problem since the types are different, but in settings where `Data` is a concrete type, it can often be that `Data = T`, so both weights and data are `T`, which could lead to accidentally getting the order wrong :thinking: 

Maybe make a struct `QuadratureParts` with named fields and return it?